ground/src/main/java/com/google/android/ground/ui/map/gms/GoogleMapsFragmentV2.kt

```diff
@@ -0,0 +1,454 @@
+  }
+
+  override fun enableGestures() = getMap().uiSettings.setAllGesturesEnabled(true)
+
+  override fun disableGestures() = getMap().uiSettings.setAllGesturesEnabled(false)
+
+  override fun moveCamera(point: Point) =
+    getMap().moveCamera(CameraUpdateFactory.newLatLng(toLatLng(point)))
+
+  override fun moveCamera(point: Point, zoomLevel: Float) =
+    getMap().moveCamera(CameraUpdateFactory.newLatLngZoom(toLatLng(point), zoomLevel))
+
+  private fun addMapPin(mapPin: MapPin) {
+    val position = toLatLng(mapPin.position)
+    // TODO: add the anchor values into the resource dimensions file
+    val marker =
+      getMap()
+        .addMarker(
+          MarkerOptions()
+            .position(position)
+            .icon(getMarkerIcon(mapPin))
+            .anchor(0.5f, 0.85f)
+            .alpha(1.0f)
+        )
+    markers.add(marker)
+    marker.tag = mapPin
+  }
+
+  private fun getMarkerIcon(mapPin: MapPin): BitmapDescriptor =
+    markerIconFactory.getMarkerIcon(parseColor(mapPin.style.color), currentZoomLevel)
+
+  private fun addMultiPolygon(locationOfInterest: MapLocationOfInterest, multiPolygon: MultiPolygon) =
+    multiPolygon.polygons.forEach { addPolygon(locationOfInterest, it) }
+
+  private fun addPolygon(locationOfInterest: MapLocationOfInterest, polygon: Polygon) {
+    val options = PolygonOptions()
+    options.clickable(false)
+    val shellVertices = polygon.shell.vertices.map { toLatLng(it) }
+    options.addAll(shellVertices)
+    val holes =
+      polygon.holes.map { hole -> hole.vertices.map { point -> toLatLng(point) } }
+    holes.forEach { options.addHole(it) }
+
+    val mapsPolygon = getMap().addPolygon(options)
+    mapsPolygon.tag = Pair(locationOfInterest.locationOfInterest!!.id, LocationOfInterest::javaClass)
+    mapsPolygon.strokeWidth = polylineStrokeWidth.toFloat()
+    // TODO(jsunde): Figure out where we want to get the style from
+    //  parseColor(Style().color)
+    mapsPolygon.fillColor = parseColor("#55ffffff")
+    mapsPolygon.strokeColor = parseColor(Style().color)
+    mapsPolygon.strokeJointType = JointType.ROUND
+
+    polygons.getOrPut(locationOfInterest) { mutableListOf()} .add(mapsPolygon)
+  }
+
+  private val polylineStrokeWidth: Int
+    get() = resources.getDimension(R.dimen.polyline_stroke_width).toInt()
+
+  private fun onMapClick(latLng: LatLng) = handleAmbiguity(latLng)
+
+  override fun getCurrentZoomLevel(): Float = getMap().cameraPosition.zoom
+
+  @SuppressLint("MissingPermission")
+  override fun enableCurrentLocationIndicator() {
+    if (!getMap().isMyLocationEnabled) {
+      getMap().isMyLocationEnabled = true
+    }
+  }
+
+  override fun setMapLocationsOfInterest(features: ImmutableSet<MapLocationOfInterest>) {
+    Timber.v("setMapLocationsOfInterest() called with ${features.size} locations of interest")
+    val features: MutableSet<MapLocationOfInterest> = HashSet(features)
+
+    //    val deletedMarkers: MutableList<Marker> = ArrayList()
```

Comment:
Intentional?

Reply:
I thought it might be helpful to keep as context for when we implement Points.